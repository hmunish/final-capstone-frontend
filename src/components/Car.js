import React from "react";
import PropTypes from "prop-types";
import styles from "../CSS/Car.module.css";
import carImg from "../assets/car-1.png";
import fb from "../assets/fb.png";
import twitter from "../assets/twitter.png";
import insta from "../assets/insta.png";

const Car = ({ name, image, description }) => (
  <div className={styles.car}>
    <div>
      <img src={carImg} alt={image} className={styles.carImg} />
    </div>
    <div className={styles.carDetailsBox}>
      <p className={styles.carName}>{name}</p>
      <p className={styles.carDescription}>{description}</p>
    </div>
    <div className={styles.carSocialBox}>
      <img src={fb} alt="Social Media Icon" />
      <img src={twitter} alt="Social Media Icon" />
      <img src={insta} alt="Social Media Icon" />
    </div>
  </div>
);

Car.propTypes = {
  name: PropTypes.string,
  image: PropTypes.string,
  description: PropTypes.string,
};

Car.defaultProps = {
  name: "",
  image: "",
  description: "",
};

export default Car;
