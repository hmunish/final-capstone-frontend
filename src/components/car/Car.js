import React from "react";
import PropTypes from "prop-types";
import { useNavigate } from "react-router-dom";
import styles from "../../CSS/Car.module.css";
import carImg from "../../assets/car-1.png";
import fb from "../../assets/fb.png";
import twitter from "../../assets/twitter.png";
import insta from "../../assets/insta.png";

const Car = ({
  id, name, image, description,
}) => {
  const navigate = useNavigate();
  const handleCarClick = (e) => {
    const { carid } = e.target.closest("#carBox")?.dataset;
    navigate(`/car/details?carId=${carid}`);
  };

  return (
    <div
      className={styles.car}
      data-carid={id}
      onClick={(e) => handleCarClick(e)}
      role="link"
      tabIndex={id}
      aria-hidden="true"
      id="carBox"
    >
      <div>
        <img src={carImg} alt={image} className={styles.carImg} />
      </div>
      <div className={styles.carDetailsBox}>
        <p className={styles.carName}>{name}</p>
        <p className={styles.carDescription}>{description}</p>
      </div>
      <div className={styles.carSocialBox}>
        <img src={fb} alt="Social Media Icon" />
        <img src={twitter} alt="Social Media Icon" />
        <img src={insta} alt="Social Media Icon" />
      </div>
    </div>
  );
};

Car.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string,
  image: PropTypes.string,
  description: PropTypes.string,
};

Car.defaultProps = {
  id: null,
  name: "",
  image: "",
  description: "",
};

export default Car;
