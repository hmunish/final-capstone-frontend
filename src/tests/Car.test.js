import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Car from '../components/car/Car';

test('Car Component matches snapshots', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Car />
    </BrowserRouter>,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
