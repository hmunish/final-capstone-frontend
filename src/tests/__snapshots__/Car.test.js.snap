// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Car Component matches snapshots 1`] = `
<div
  aria-hidden="true"
  className="car"
  data-carid={null}
  id="carBox"
  onClick={[Function]}
  role="link"
  tabIndex={null}
>
  <div>
    <img
      alt="Car"
      className="carImg"
      src=""
    />
  </div>
  <div
    className="carDetailsBox"
  >
    <p
      className="carName"
    />
    <p
      className="carDescription"
    />
  </div>
  <div
    className="carSocialBox"
  >
    <img
      alt="Social Media Icon"
      src="fb.png"
    />
    <img
      alt="Social Media Icon"
      src="twitter.png"
    />
    <img
      alt="Social Media Icon"
      src="insta.png"
    />
  </div>
</div>
`;
